### ISSUE TITLE
Add global contexts features

### ISSUE THREAD

**Issue by fblundun** [snowplow/snowplow-javascript-tracker#405]
e.g.

``` js
snowplow('registerCommonContext', myContext);
snowplow('registerCommonContext', myContextGeneratingFunction);
```

**Comment by alexanderdean** [comment]
I like!

How about:

```javascript
snowplow('attachContext', '*', myContext);
snowplow('attachContext', myFilterFunction, myContextGeneratingFunction);
```

**Comment by fblundun** [comment]
What does myFilterFunction filter?

**Comment by alexanderdean** [comment]
If myFilterFunction returns true, attach the context. TBC the arguments passed to myFilterFunction.

**Comment by alexanderdean** [comment]
@yalisassoon, @bogaert - feedback please!

**Comment by yalisassoon** [comment]
I love it.

When users implement the JS tracker through GTM, a common pattern is to have some JS at the top of the page that constructs the contexts array out of the different objects in the `dataLayer`. Pseudocode looks something like this:

* Create empty contexts array
* If `dataLayer` array contains `user` object add schema to it and push it onto contexts array
* If `dataLayer` array contains `product` object add schema to it and push it onto the contexts array
* ...

Then add the context to any subsequent `track` methods called in the tag.

So anything that simplifies the process of building the contexts array prior to firing the `track` methods is useful.

**Comment by alexanderdean** [comment]
Bringing forwards

**Comment by yalisassoon** [comment]
Team at Snowflake Analytics has expressed an interest in this. (And equivalent functionality in Objective C and Android trackers.)

**Comment by alexanderdean** [comment]
See also: https://github.com/snowplow/snowplow/issues/3852

**Comment by pavelivanov** [comment]
Hi there. There is "Global contexts" in the Snowplow 2.10.0 documentation - https://github.com/snowplow/snowplow/wiki/2-Specific-event-tracking-with-the-Javascript-tracker

I checked 2.10.0-rc1 for "addGlobalContexts" event but didn't find it or other global context functionality. 

Will "Global contexts" be implemented in 2.10.0?
